### ISSUE TITLE
[Suggestion]Let Monika teach player how to play chess

### ISSUE THREAD

**Issue by Wingdinggaster656** [Monika-After-Story/MonikaModDev#7221]
Okay, I know Monika said she is still not that great to teach player piano, but I think her character should be able to teach player a little chess. After all, this character is supposed to be "pretty well" at chess.
I don't expect Monika to tell you all the advanced things about chess, but she can at least teach you the basic principle of chess, right?
So let us add chess tutorial in MAS like how Python Tip take places in MAS. Player can ask some questions like:
1."How to move a pawn?"
2."How to move a bishop"
3."How to move a..."
You could even ask Monika to teach you something else, like a few key points of chess, like the goal of the opening is to control center, and give some examples of why.

If you guys think this is okay with Monika the character, then fine, I guess I can do this job.

**Comment by Ant9cp** [comment]
Oh great let's hope this is possible

**Comment by multimokia** [comment]
The new chess displayable introduced in #6163 will be of use for this, given you can request highlights to draw attention to specific points on the board, along with control the pieces from both sides *without* the engine. You can see an example of it [here](https://cdn.discordapp.com/attachments/393978061084688405/742788939794088159/2020-08-11_12-53-36.mp4).

We mainly just needed someone who more or less knew their way around chess more than we did to get them done

**Comment by Wingdinggaster656** [comment]
> The new chess displayable introduced in #6163 will be of use for this, given you can request highlights to draw attention to specific points on the board, along with control the pieces from both sides _without_ the engine. You can see an example of it [here](https://cdn.discordapp.com/attachments/393978061084688405/742788939794088159/2020-08-11_12-53-36.mp4).
> 
> We mainly just needed someone who more or less knew their way around chess more than we did to get them done

Thanks for replying me. Hearing your reply, I'm sure I can do this job now.
My chess level-rating is 2100(Grandmaster are 2400 or higher, people are probably 1300 around), which means I'm better at chess than almost most people, I might even become a grandmaster someday. If you don't mind, I'd like to give player chess lessons with questions to ask like python tips.
I think I will probably give you a finished product this week.

**Comment by Wingdinggaster656** [comment]
https://user-images.githubusercontent.com/84446131/119003814-9e8f8f80-b9c0-11eb-8c9d-9b822016f733.mp4
@multimokia 
Sorry for being late, I just went to do some homework. I have finished 4 lessons now, this is one of them. Others are something like this one. Is this okay?

**Comment by multimokia** [comment]
It's not bad, but I think a key thing to go into is *why* she moves where she does, and slow it down more so it's easier for the player, and especially those who are new to chess, to follow what she's doing and why she's doing it.

It's one thing to show what's happening, but to actually make it a demonstration which provides understanding is another.

That's the thing I think we need to do with the chess lessons thing, that being to make it like a full kinda intro to chess that she teaches.

**Comment by Wingdinggaster656** [comment]
> It's not bad, but I think a key thing to go into is _why_ she moves where she does, and slow it down more so it's easier for the player, and especially those who are new to chess, to follow what she's doing and why she's doing it.
> 
> It's one thing to show what's happening, but to actually make it a demonstration which provides understanding is another.
> 
> That's the thing I think we need to do with the chess lessons thing, that being to make it like a full kinda intro to chess that she teaches.

Oh, then this is not worrying. This checkmate is the famous "Two Rooks Checkmate", I made a special lesson to introduce this.
All things that you saw she didn't explain is because she explained them in other lessons, don't worry. I'm a chess coach, and have taught many people how to play chess.
Since I think your words are saying "This idea is okay", I am going to finish the resting about 40 lessons now. I will give you the final product soon.

**Comment by multimokia** [comment]
Definitely think the first 8 should be skippable, considering people likely played with her and got movements a bit down.

That or in general come into it knowing how to move pieces

**Comment by multimokia** [comment]
Actually we can also probably condense how to move pieces into a single lesson. But ideally that'd also still be skippable

**Comment by Wingdinggaster656** [comment]
@multimokia
Well, it seems that I overestimated my production speed, maybe this thing will take me a long time......
In short, I added the "player chooses to move" factor to the course, and now Monika will stop at key points, ask the player's opinion, and analyze it, etc....... The result was even only one single lesson costs 800 lines of codes.
In short, I will definitely work on it again, but not as quickly as I expected...... I've written seven lessons so far, and since most of the individual lessons game play video are longer than seven or eight minutes (including time for players to understand, or three or four minutes if they just read), the final video file size is over 10 MBs, which means I can't send preview videos here anymore.
But I will come here from time to time to announce the progress of my production, please don't worry. I'll finish this as soon as possible...... Gosh, I estimate that the final product will take tens of thousands of lines of code.

**Comment by Wingdinggaster656** [comment]
Oh, and, I don't know if you guys found my another issue. Can I add numbers and letters on the board image so player can read the board better? This is almost everywhere doing.

**Comment by multimokia** [comment]
You can open a pr with the modified board and we'll review it

**Comment by Wingdinggaster656** [comment]
Progress report 2021-05-26:
I have finished the basic lessons to lead player to the level-rating about 1100, finished 11 lessons so far.
The number of lines of code so far is over 3000. The final result would be around 10000 I guess.
But I am very sorry that I have been very busy in the last two weeks, so I can't concentrate on this. Still, I promise I'll get it done, or at least most of it, by next month. For the next two weeks, I should be able to maintain the pace of completing one class a week.

**Comment by Booplicate** [comment]
I think it would be way too hard to review. Let's split it into parts. Start with the main event that unlocks first lessons and a few basic lessons. Once we're sure the structure works fine, we can always add more lessons later.

**Comment by Wingdinggaster656** [comment]
> I think it would be way too hard to review. Let's split it into parts. Start with the main event that unlocks first lessons and a few basic lessons. Once we're sure the structure works fine, we can always add more lessons later.

You are right, but for "basic lessons", there is still one or two lesson I have not finished yet. It's still gonna have to wait about two weeks, sorry for this.

**Comment by Booplicate** [comment]
That's totally fine, there's no rush.

**Comment by Wingdinggaster656** [comment]
The work-process-report of this issue has been transferred to this Pull Request:
#7259

If you have any good ideas, please leave a comment there, not here. I will close this Issue after that Pull Request is completed.

**Comment by MASsuperfan** [comment]
Is this Submod ready yet by any chance? I would really like to try it out.

**Comment by Booplicate** [comment]
This doesn't appear to be a submod.

**Comment by Wingdinggaster656** [comment]
> Is this Submod ready yet by any chance? I would really like to try it out.

This is not a Submod. At least not planned as.

And then yes, these contents are almost ready. They just need some other projects to be completed first (such as adjusting the difficulty of chess).

**Comment by MASsuperfan** [comment]
> > Is this Submod ready yet by any chance? I would really like to try it out.
> 
> This is not a Submod. At least not planned so.
> 
> And then yes, these contents are almost ready. They just need some other projects to be completed first (such as adjusting the difficulty of chess).

Okay then, thank you, i just don't know chess and can't seem to learn it lol.
